public/app/features/library-panels/components/LibraryPanelsSearch/LibraryPanelsSearch.tsx

```diff
@@ -133,34 +98,118 @@ export const LibraryPanelsSearch = ({
   );
 };
 
-function getStyles(theme: GrafanaTheme2) {
+function getStyles(theme: GrafanaTheme2, variant: LibraryPanelsSearchVariant) {
+  const tightLayout = css`
+    flex-direction: row;
+    row-gap: ${theme.spacing(1)};
+  `;
   return {
+    filterInputWrapper: css`
+      flex-grow: ${variant === LibraryPanelsSearchVariant.Tight ? 1 : 'initial'};
+    `,
     container: css`
       width: 100%;
       overflow-y: auto;
       padding: ${theme.spacing(1)};
     `,
-    buttonRow: css`
-      display: flex;
-      justify-content: space-between;
+    libraryPanelsView: css`
       width: 100%;
-      margin-top: ${theme.spacing(2)}; // Clear types link
     `,
-    tightButtonRow: css`
+    gridContainer: css`
+      ${variant === LibraryPanelsSearchVariant.Tight ? tightLayout : ''};
       display: flex;
-      justify-content: space-between;
-      width: 100%;
-      margin-top: ${theme.spacing(4)}; // Clear types link
-    `,
-    tightFilter: css`
-      flex-grow: 1;
-    `,
-    tightSortFilter: css`
-      flex-grow: 1;
-      padding: ${theme.spacing(0, 0, 0, 0.5)};
-    `,
-    libraryPanelsView: css`
+      flex-direction: column;
       width: 100%;
+      column-gap: ${theme.spacing(1)};
+      row-gap: ${theme.spacing(1)};
+      padding-bottom: ${theme.spacing(2)};
     `,
   };
 }
+
+interface SearchControlsProps {
+  showSort: boolean;
+  showPanelFilter: boolean;
+  showFolderFilter: boolean;
+  sortDirection?: string;
+  onSortChange: (sortValue: SelectableValue) => void;
+  onFolderFilterChange: (folder: string[]) => void;
+  onPanelFilterChange: (plugins: string[]) => void;
+  variant?: LibraryPanelsSearchVariant;
+}
+
+const SearchControls = React.memo(
+  ({
+    variant = LibraryPanelsSearchVariant.Spacious,
+    showSort,
+    showPanelFilter,
+    showFolderFilter,
+    sortDirection,
+    onSortChange,
+    onFolderFilterChange,
+    onPanelFilterChange,
+  }: SearchControlsProps) => {
+    const styles = useStyles2(useCallback((theme) => getRowStyles(theme, variant), [variant]));
+    const panelFilterChanged = useCallback(
+      (plugins: PanelPluginMeta[]) => onPanelFilterChange(plugins.map((p) => p.id)),
+      [onPanelFilterChange]
+    );
+    const folderFilterChanged = useCallback(
+      (folders: FolderInfo[]) => onFolderFilterChange(folders.map((f) => String(f.id))),
+      [onFolderFilterChange]
+    );
+
+    return (
+      <div className={styles.container}>
+        {showSort && <SortPicker value={sortDirection} onChange={onSortChange} filter={['alpha-asc', 'alpha-desc']} />}
```

Comment:
Should we extract the filter strings to an enum here? WDYT?

Reply:
It is indeed not otherwise obvious that those two are part of [the options `SortPicker` has](https://github.com/grafana/grafana/blob/8af587e7bae589820da0a7e459ace1274af3619c/public/app/core/components/Select/SortPicker.tsx#L28) after receiving its response from the api `/api/search/sorting`.